src/sfctl/custom_cluster.py

```diff
@@ -47,32 +53,84 @@ def select(endpoint, cert=None, key=None, pem=None, ca=None, no_verify=False):
     :param str pem: Path to client certificate, as a .pem file
     :param str ca: Path to CA certs directory to treat as valid or CA bundle
     file
+    :param bool aad: Indicate using Azure Active Directory authentication
     :param bool no_verify: Disable verification for certificates when using
     HTTPS, note: this is an insecure option and should not be used for
     production environments
     """
-    from sfctl.config import (set_ca_cert, set_cert, set_cluster_endpoint,
-                              set_no_verify)
+    from sfctl.config import (set_ca_cert, set_auth,
+                              set_cluster_endpoint, client_endpoint,
+                              set_no_verify, no_verify_setting)
     from msrest import ServiceClient, Configuration
-    from sfctl.auth import ClientCertAuthentication
+    from sfctl.auth import ClientCertAuthentication, AdalAuthentication
 
-    select_arg_verify(endpoint, cert, key, pem, ca, no_verify)
+    select_arg_verify(endpoint, cert, key, pem, ca, aad, no_verify)
 
-    client_cert = None
-    if pem:
-        client_cert = pem
-    elif cert:
-        client_cert = (cert, key)
+    access_token = None
+    if aad:
+        endpoint_backup = client_endpoint()
```

Comment:
You don't need to perform these backups if you create the python SDK client manually rather than calling the apiclient.